bundle/QueryType/SearchQueryType.php

```diff
@@ -27,38 +31,77 @@ public static function getName(): string
 
     protected function configureOptions(OptionsResolver $optionsResolver): void
     {
-        $optionsResolver->setRequired(['search_text', 'content_types', 'subtree']);
-
+        $optionsResolver->setRequired(['search_text', 'content_types', 'subtree', 'sort', 'order']);
         $optionsResolver->setAllowedTypes('search_text', 'string');
         $optionsResolver->setAllowedTypes('content_types', 'string[]');
+        $optionsResolver->setAllowedTypes('sort', 'string[]');
         $optionsResolver->setAllowedTypes('subtree', ['int', 'string']);
-
         $optionsResolver->setAllowedValues(
             'search_text',
             static fn (string $searchText): bool => trim($searchText) !== '',
         );
-
+        $optionsResolver->setAllowedValues(
+            'sort',
+            /** @var string[] $keys */
+            fn (array $keys): bool => $this->sortKeysAllowed($keys),
+        );
+        $optionsResolver->setAllowedValues(
+            'order',
+            static fn (string $searchText): bool => trim($searchText) === 'asc' || trim($searchText) === 'desc',
+        );
         $optionsResolver->setDefault('content_types', $this->configResolver->getParameter('search.content_types', 'ngsite'));
         $optionsResolver->setDefault('subtree', $this->site->getSettings()->rootLocationId);
+        $optionsResolver->setDefault('sort', [Query\SortClause\DatePublished::class]);
+        $optionsResolver->setDefault('order', 'desc');
     }
 
     protected function doGetQuery(array $parameters): Query
     {
         $subtreeLocation = $this->site->getLoadService()->loadLocation($parameters['subtree']);
-
+        $sortingKeys = $parameters['sort'];
+        $order = $parameters['order'];
+        $descendingOrder = $order === 'desc';
```

Comment:
Why do we need `$descendingOrder` variable? Can't you just use `$parameters['order']` direcly, and make their values correspond to `Query::SORT_DESC` and `Query::SORT_ASC`, so below you can just use `new $name($parameters['order'])`?

Reply:
Fixed!